### PR TITLE
Update config files for external hooks

### PR DESCRIPTION
**Pull request by lilyminium**
The general updates here are:
* adding a pre-commit configuration. I think this could be very good for new developers who don't want to deal with formatting themselves. I've gone with flake8 over black due to generally more options.
* Line length is set to 80 characters everywhere.
* Added pylint config file

### CODE REVIEW

File: {{cookiecutter.repo_name}}/readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# readthedocs.yaml
+
+version: 2
+
+build:
+  os: ubuntu-20.04
```
Comment (IAlibay) [inline {{cookiecutter.repo_name}}/readthedocs.yaml:6]: do they let you do latest here? this would lower our maintenance burdens

Comment (lilyminium) [reply {{cookiecutter.repo_name}}/readthedocs.yaml:6]: I swear I used to be able to, but I think we're restricted now to 20.04 or 22.04 https://docs.readthedocs.io/en/stable/config-file/v2.html#build-os

Comment (IAlibay) [reply {{cookiecutter.repo_name}}/readthedocs.yaml:6]: that's a pain, could we use 22.04 here at least so we get 4 years of blissfully not having to think about it?

**Review by IAlibay — APPROVED** [review]
lgtm

**Pull request merged**